### ISSUE TITLE
Plugin incompatible with pedantic? abort

### ISSUE THREAD

**Issue by marcobiscaro2112** [clojure-lsp/lein-clojure-lsp#7]
If I add the plugin in a new project that uses 
```
:pedantic? :abort
```
the dependency resolution will fail even if the project itself doesn't have dependencies:
```
Possibly confusing dependencies found:
[com.github.clojure-lsp/lein-clojure-lsp "1.3.24"] -> [com.github.clojure-lsp/clojure-lsp-server "2023.07.01-22.35.41"] -> [org.clojure/tools.cli "1.0.214"]
 overrides
[com.github.clojure-lsp/lein-clojure-lsp "1.3.24"] -> [com.github.clojure-lsp/clojure-lsp-server "2023.07.01-22.35.41"] -> [dev.weavejester/cljfmt "0.10.6" :exclusions [rewrite-cljs com.googlecode.java-diff-utils/diffutils]] -> [org.clojure/tools.cli "1.0.219"]

Consider using these exclusions:
[com.github.clojure-lsp/lein-clojure-lsp "1.3.24" :exclusions [org.clojure/tools.cli]]

[com.github.clojure-lsp/lein-clojure-lsp "1.3.24"] -> [com.github.clojure-lsp/clojure-lsp-server "2023.07.01-22.35.41"] -> [com.github.clojure-lsp/lsp4clj "1.8.0" :exclusions [funcool/promesa]] -> [cheshire "5.11.0"] -> [com.fasterxml.jackson.core/jackson-core "2.13.3"]
 overrides
[com.github.clojure-lsp/lein-clojure-lsp "1.3.24"] -> [com.github.clojure-lsp/clojure-lsp-server "2023.07.01-22.35.41"] -> [clj-kondo "2023.06.02-20230630.144012-16"] -> [com.cognitect/transit-clj "1.0.333"] -> [com.cognitect/transit-java "1.0.371"] -> [com.fasterxml.jackson.core/jackson-core "2.14.2"]
 and
[com.github.clojure-lsp/lein-clojure-lsp "1.3.24"] -> [com.github.clojure-lsp/clojure-lsp-server "2023.07.01-22.35.41"] -> [com.cognitect/transit-clj "1.0.333"] -> [com.cognitect/transit-java "1.0.371"] -> [com.fasterxml.jackson.core/jackson-core "2.14.2"]

Consider using these exclusions:
[com.github.clojure-lsp/lein-clojure-lsp "1.3.24" :exclusions [com.fasterxml.jackson.core/jackson-core]]

Aborting due to :pedantic? :abort
```

Ideally `com.github.clojure-lsp/lein-clojure-lsp` would not pull any possibly conflicting transitive dependency.

**Comment by marcobiscaro2112** [comment]
Should be trivial to fix with managed deps. Something like this in lein-clojure-lsp fixes the problem:

```diff
diff --git a/project.clj b/project.clj
index d73ff39..889bd89 100644
--- a/project.clj
+++ b/project.clj
@@ -4,9 +4,11 @@
   :license {:name "MIT"
             :url "https://opensource.org/licenses/MIT"}
   :eval-in-leiningen true
-  :pedantic? :warn
+  :pedantic? :abort
   :deploy-repositories [["clojars" {:url           "https://clojars.org/repo"
                                     :username      :env/clojars_username
                                     :password      :env/clojars_password
                                     :sign-releases false}]]
+  :managed-dependencies [[com.fasterxml.jackson.core/jackson-core "2.15.2"]
+                         [org.clojure/tools.cli "1.0.219"]]
   :dependencies [[com.github.clojure-lsp/clojure-lsp-server "2023.07.01-22.35.41"]])
```

Note that this does _not_ work in the project that uses lein-clojure-lsp, because managed deps don't apply to plugins.

**Comment by ericdallo** [comment]
@marcobiscaro2112 looks good, willing to open a PR with those changes?

**Comment by marcobiscaro2112** [comment]
@ericdallo done in #8.

**Comment by ericdallo** [comment]
Thanks, released as 1.4.0
